app/plugin.go

```diff
@@ -15,22 +15,49 @@ import (
 	"github.com/mattermost/mattermost-server/utils"
 )
 
+// GetPluginsEnvironment returns the plugin environment for use if plugins are enabled and
+// initialized.
+//
+// To get the plugins environment when the plugins are disabled, manually acquire the plugins
+// lock instead.
+func (a *App) GetPluginsEnvironment() *plugin.Environment {
+	if !*a.Config().PluginSettings.Enable {
+		return nil
+	}
+
+	a.Srv.PluginsLock.RLock()
+	defer a.Srv.PluginsLock.RUnlock()
+
+	return a.Srv.PluginsEnvironment
+}
+
+func (a *App) SetPluginsEnvironment(pluginsEnvironment *plugin.Environment) {
+	a.Srv.PluginsLock.Lock()
+	defer a.Srv.PluginsLock.Unlock()
+
+	a.Srv.PluginsEnvironment = pluginsEnvironment
+}
+
 func (a *App) SyncPluginsActiveState() {
-	if a.Srv.Plugins == nil {
+	a.Srv.PluginsLock.RLock()
```

Comment:
Methods in this file use `PluginsLock` directly, because they need access to the plugins environment even if it isn't enabled.